File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/UXSeparator.php
<?php
namespace php\gui;

/**
 * Class UXSeparator
 * @package php\gui
 * @packages gui, javafx
 */
class UXSeparator extends UXControl
{
    /**
     * @var string HORIZONTAL or VERTICAL
     */
    public $orientation = 'HORIZONTAL';

    /**
     * @var string
     */
    public $hAlignment;

    /**
     * @var string
     */
    public $vAlignment;

    /**
     * UXSeparator constructor.
     * @param string $orientation
     */
    public function __construct($orientation = 'HORIZONTAL')
    {
    }
}

File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/UXSeparator.java
package org.develnext.jphp.ext.javafx.classes;

import javafx.geometry.HPos;
import javafx.geometry.Orientation;
import javafx.geometry.VPos;
import javafx.scene.control.Separator;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Reflection.Name(JavaFXExtension.NS + "UXSeparator")
public class UXSeparator extends UXControl<Separator> {
    interface WrappedInterface {
        @Property Orientation orientation();
        @Property("hAlignment") HPos halignment();
        @Property("vAlignment") VPos valignment();
    }

    public UXSeparator(Environment env, Separator wrappedObject) {
        super(env, wrappedObject);
    }

    public UXSeparator(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void __construct() {
        __wrappedObject = new Separator();
    }

    @Signature
    public void __construct(Orientation orientation) {
        __wrappedObject = new Separator(orientation);
    }
}
